src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -1,44 +1,104 @@
-cd "$SOURCE_DIR"
-
 # Yarn config is per user, and since the build might run with a non-root account, we make sure
 # the yarn cache is set on every build.
 YARN_CACHE_DIR=/usr/local/share/yarn-cache
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
-    echo "Configuring Yarn cache folder ..."
+    echo "Configuring Yarn cache folder..."
     yarn config set cache-folder $YARN_CACHE_DIR
 fi
 
+allNodeModulesDirName=__oryx_all_node_modules
+if [ -d "$SOURCE_DIR"/$allNodeModulesDirName ]
```

Comment:
Since base bash script copies the content from source directory to destination directory, I wanted the source directory to have the content that we actually want to be in the destination directory. Since we do not want all the modules to be copied to destination directory, we rename this folder and exclude it from rsync.

Reply:
we already leave node_modules folder out when copying the files from source to target